embedder/embedder_helper.c

```diff
@@ -25,6 +25,29 @@ void proxy_desktop_binary_reply(const uint8_t *data, size_t data_size,
                                 void *user_data);
 
 // C helper
+
+void loadSnapshot(FlutterProjectArgs *Args, char* vm_snapshot_data_path, char* vm_snapshot_instructions_path, char* isolate_snapshot_data_path, char* isolate_snapshot_instructions_path) {
+    FILE *vm_snapshot_data_file = fopen(vm_snapshot_data_path, "r");
+    FILE *vm_snapshot_instructions_file = fopen(vm_snapshot_instructions_path, "r");
+    FILE *isolate_snapshot_data_file = fopen(isolate_snapshot_data_path, "r");
+    FILE *isolate_snapshot_instructions_file = fopen(isolate_snapshot_instructions_path, "r");
+
+    Args->vm_snapshot_data = (uint8_t*) vm_snapshot_data_file;
+    Args->vm_snapshot_instructions = (uint8_t*) vm_snapshot_instructions_file;
+    Args->isolate_snapshot_data = (uint8_t*) isolate_snapshot_data_file;
+    Args->isolate_snapshot_instructions = (uint8_t*) isolate_snapshot_instructions_file;
+
+    Args->vm_snapshot_data_size = sizeof Args->vm_snapshot_data;
+    Args->vm_snapshot_instructions_size = sizeof Args->vm_snapshot_instructions;
+    Args->isolate_snapshot_data_size = sizeof Args->isolate_snapshot_data;
+    Args->isolate_snapshot_instructions_size = sizeof Args->isolate_snapshot_instructions;
+
+    fclose(vm_snapshot_data_file);
```

Comment:
dumb question as I'm not really a go developer. You are setting a pointer in `Args->vm_snapshot_data` and then closing the file, won't that pointer be dangling?

Reply:
Your watching C code, but let me try

Reply:
Nope didn't work